### DISCUSSION TITLE
CLI and Flags 🗣️🇰🇷

### DISCUSSION THREAD

**Discussion by hahwul** [owasp-noir/noir#421] [Ideas]
> [!NOTE]  
We used Korean for communication among operators. If you have any comments or suggestions, feel free to leave them in English.

@ksg97031 
Noir가 기능이 점점 많아지다 보니 CLI 사용과 Flag에 대한 고민이 듭니다.
현재는 `noir `로 사용하고 있지만, 이는 기능 표현에 분명한 한계가 존재합니다. 그래서 서브 커맨드를 사용할지 방향을 잡아야 할 것 같습니다.

e.g
* `noir scan <BASE> <FLAGS>`
* `noir shell-completion`

현재는 소스코드에 대한 스캔이 기본이라 아직 필요하다고 느끼지는 않으나, completion 이외에도 점차 기능이 넓어진다면, 고려해야 할 수도 있습니다.
다만 이 선택을 한다면 기존 버전과 호환이 불가능하기 때문에 우리는 아마도 Major 버전(to 1. x1.x)을 올려야 할 것 같네요.

이 선택은 앞으로의 방향성을 결정할 수 있는 선택이고 신중해야 하기에 우선 Discussions에 남겨둡니다.

**Comment by ksg97031** [comment]
@hahwul 
주요 기능 확장에 따라 서브 커맨드 분리가 필요할 것 같네요.
인지하고 있겠습니다~
